include/swift/AST/ASTContext.h

```diff
@@ -168,6 +168,7 @@ typedef llvm::PointerUnion<NominalTypeDecl *, ExtensionDecl *>
 class ASTContext {
   ASTContext(const ASTContext&) = delete;
   void operator=(const ASTContext&) = delete;
+  FuncDecl *ArrayAppendElementDecl;
```

Comment:
Please follow the approach used e.g. by `ASTContext::getEqualIntDecl` to lookup a function and cache it. Also use a similar naming scheme, i.e. `getArrayAppendElementDecl`.